pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -799,12 +799,11 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 			}
 		} else if pod != nil {
 			// for split horizon EDS k8s multi cluster, in case there are pods of the same ip across clusters,
-			// which can happen when multi clusters using same pod cidr.
-			// As we have proxy Network meta, compare it with the network which endpoint belongs to,
-			// if they are not same, ignore the pod, because the pod is in another cluster.
-			if proxy.Metadata.Network != c.endpointNetwork(proxyIP) {
+			// which can happen when multi clusters using same pod cidr. ignore the pod if it isn't in this cluster.
+			if proxy.Metadata.ClusterID != c.clusterID {
```

Comment:
This can not happen

Reply:
Are you saying the existing guard is unnecessary?  Or would they behave differently?

Reply:
I mean this(clusterID not equal) should not happen

Reply:
Ignore above.  looks good